### PR TITLE
Introduce `Planet` and separate the action obsoletions by them

### PR DESCRIPTION
**Pull request by longfin**
---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1205745461458878

### CODE REVIEW

File: Lib9c/Action/ActionObsoleteAttributeExtension.cs [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+                {
+                    return true;
+                }
+
+                // Comparison with ObsoleteIndex + 2 is intended to have backward compatibility with
+                // a bugged original implementation.
+                return attrs.Single().ObsoleteIndex + 2 <= blockIndex;
+            }
+
+            return attrs.Any(attr => attr.ObsoleteIndex <= blockIndex && attr.AffectedPlanet == planet);
```
Comment (moreal) [inline Lib9c/Action/ActionObsoleteAttributeExtension.cs:29]: Does it need to check `ObsoleteIndex + 2 <= blockIndex` too when the `attr.AffectedPlanet` is `Planet.Odin`?

**Review by moreal — COMMENTED** [review]
It needs follow-up PR.

File: Lib9c/Action/ActionObsoleteAttributeExtension.cs (hunk 1 above)
Comment (longfin) [reply Lib9c/Action/ActionObsoleteAttributeExtension.cs:29]: AFIK, +2 offset was needed to aide previous bug and it doesn't be needed against new chains. am I understanding correctly? @greymistcube

Comment (moreal) [reply Lib9c/Action/ActionObsoleteAttributeExtension.cs:29]: Yes, `Planet.Heimdall` and `Heimdall.Idun` will not need the +2 offset. But I think `Planet.Odin` needs +2 offset as I said. Odin is not a new chain.

Comment (longfin) [reply Lib9c/Action/ActionObsoleteAttributeExtension.cs:29]:
```suggestion
            return attrs.Any(attr => (attr.AffectedPlaent == Planet.Odin ? attr.ObsoleteIndex + 2 : attr.ObsoleteIndex) <= blockIndex && attr.AffectedPlanet == planet);
```

@moreal Ah, I see. you mean this?

Comment (moreal) [reply Lib9c/Action/ActionObsoleteAttributeExtension.cs:29]: @longfin Ah yes!! That is what I mean.

**Review by moreal — APPROVED** [review]

**Pull request merged**